recommend/recommend.go

```diff
@@ -83,23 +94,83 @@ func Recommend(c *k8s.Client, o Options) error {
 		ReportInit(o.ReportFile)
 	}
 
-	o.Images = unique(o.Images)
+	if len(o.Images) == 0 {
+		// recommendation based on k8s manifest that
+		// matches the given namespace and labels
+		if o.Namespace == "" {
+			return errors.New("k8s policy recommendation needs namespace option. Try with --namespace option.")
+		}
+
+		dps, err := c.K8sClientset.AppsV1().Deployments(o.Namespace).List(context.TODO(), v1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		for _, dp := range dps.Items {
+			selectorLabels := []string{}
+			for k, v := range dp.Spec.Selector.MatchLabels {
+				selectorLabels = append(selectorLabels, fmt.Sprintf("%s=%s", k, v))
+			}
+
+			matched := true
+			for _, filterLabel := range o.Labels {
+				if !slices.Contains(selectorLabels, filterLabel) {
+					matched = false
+					break
+				}
+			}
```

Comment:
There is no need to create another array here.
Since MatchLabels is a map indexed by label key, you can directly use the input key to check whether the label exists and compare it.
Also it would be good to have a function like `matchLabels()` used here so that the main logic does not look so complicated.

Reply:
Done.